src/compiler/transformers/test/map-imports-to-path-aliases.spec.ts

```diff
@@ -0,0 +1,35 @@
+import { mockValidatedConfig } from '@stencil/core/testing';
+import { ValidatedConfig } from '../../../internal';
+import { transpileModule } from './transpile';
+
+describe('mapImportsToPathAliases', () => {
+  let module: ReturnType<typeof transpileModule>;
+  let config: ValidatedConfig;
+
+  beforeEach(() => {
+    config = mockValidatedConfig({ tsconfig: './tsconfig.json', tsCompilerOptions: {} });
+  });
+
+  /**
+   * This test fails
+   * The `transpileModule` helper method is designed to only transpile a single module's (file's)
+   * content. However, with the solution implemented to fix module imports not getting transformed respective of
+   * their path aliases in the `tsconfig.json` file for a project, this helper cannot resolve the defined imports so
+   * the text returned from the helper keeps the original import path intact.
+   */
+  it('should replace the path alias with the generated relative path', () => {
```

Comment:
Failing test. The testing helpers we have do not account for the need to test with multiple JS modules.

Reply:
So, this is tricky. We need to restore some functionality where we have a compiler instance along with a virtual filesystem of some kind, and a patched TS compiler (all for testing things in a more 'e2e' fashion).  In the mean time, for unit testing, does spying on this TS method make things go away? TAL at the patch I attached to this comment (GH doesn't support '.patch' extensions, go figure 🙄). But you should be able to apply it with `git apply` all the same.

[ts_spy.txt](https://github.com/ionic-team/stencil/files/9338991/ts_spy.txt)

I didn't take a super close look here just yet, but this should unblock you at the very least on the testing front

Reply:
@rwaskiewicz I can do something like that, but at that point I question if that test really adds any benefit since it's not actually performing the action we're relying on TS for 🤷‍♂️ The test doesn't blow up, it just fails cuz the text doesn't get replaced like it does in a build.

With that said, we won't get "correct" test coverage either way until STENCIL-223 is fixed. So, I'm curious how we wanna handle this. Feels like we'd either need a dev build to have this verified by users (and framework team) or some kind of config flag. I know we had considered this being a potential breaking change too

Reply:
Fair points! I'll get those tests and some general code comments added in another commit. I'll also add a "todo" to get the actual module resolution step tested once STENCIL-223 is done